lmwrapper/huggingface_wrapper.py

```
@@ -175,13 +302,250 @@ def _gather_logprobs_from_logits(
+    elif runtime == Runtime.ORT_TENSORRT:
+        if torch_device.type != "cuda":
+            msg = (
+                "Cannot run model on CUDA without CUDA. Please specify"
+                " device='cuda'."
+            )
+            raise Exception(
+                msg,
+            )
+
+        provider_options = {
+            "trt_engine_cache_enable": True,
+            "trt_engine_cache_path": (
+                f"tmp/trt_cache_{model_name.replace('/','_')}_tensorrt"
+            ),
+        }
+
+        # TensorRT models do not support these flags
+        _kwargs.pop("low_cpu_mem_usage", None)
+        _kwargs.pop("device_map", None)
+
+        model = get_ort_model(model_class).from_pretrained(
+            model_name,
+            export=True,
+            provider="TensorrtExecutionProvider",
+            provider_options=provider_options,
+            session_options=session_options,
+            **_kwargs,
+        )
+        assert "TensorrtExecutionProvider" in model.providers
+    elif runtime == Runtime.BETTER_TRANSFORMER:
+        model = BetterTransformer.transform(
+            model_class.from_pretrained(
+                model_name,
+                device_map="auto",
+                torch_dtype=precision,
+                **_kwargs,
+            ),
+        )
+    else:
+        msg = "Invalid Runtime provided."
+        raise Exception(msg)
+
+    predictor = HuggingfacePredictor(tokenizer, model, device=torch_device)
+
+    if runtime == Runtime.ORT_TENSORRT:
+        # Warm up TensorRT model once instantiated.
+        print("Warmimg up TensorRT model.")
+        warmup_model(predictor)
+        print("Warmup successful.")
+
+    return predictor
+
+
+def warmup_model(predictor: HuggingfacePredictor):
+    small_prompt = LmPrompt("!", cache=False, temperature=0)
+    predictor.predict(small_prompt)
+
+    long_prompt_str = (
+        "hello world" * predictor.get_model_max_length()
+    )  # TODO: this should be valid model tokens!
```

I'd raise a NotImplemented error in this module. This will almost certainly fail with an inscrutable error message since the "hello world" likely becomes multiple tokens. Until we add this part probably best to guard with NotImplemented error rather than only the TODO comment in this context.

---

Agreed. Originally I had truncation enabled for the tokenizer but removed that since I agree we should not change defaults that cause unexpected behavior.